constantine/commitments/eth_verkle_ipa.nim

```diff
@@ -139,7 +143,7 @@ func innerProduct[F](r: var F, a, b: distinct(View[F] or MutableView[F])) =
 func ipa_commit*[N: static int, EC, F](
       crs: PolynomialEval[N, EC],
       r: var EC,
-      poly: PolynomialEval[N, F]) =
+      poly: PolynomialEval[N, F]) {.libPrefix: prefix_ipa.} =
```

Comment:
This is not the proper level of exports as this is a generic procedure.

Export should happen at fully specified protocol level (i.e. curve, field, hash functions, `N: static int` input sizes are known).

In this case, this means having proper instantiated `ipa_commit` here: 
https://github.com/mratsim/constantine/blob/2c3de8738d031394dfb78e3fc839cf5d12e42b92/constantine/ethereum_verkle_ipa.nim#L285-L292

and tagging those with libPrefix.

Reply:
is there any example of a wrapper i can follow to understand?

Reply:
Yes, anything with generics cannot be exported and need a concretely typed wrapper.